couch/check.py

```diff
@@ -186,14 +186,88 @@ def _build_metrics(self, data, tags, prefix = 'couchdb'):
             elif type(value) is dict:
                 self._build_metrics(value, tags, "{0}.{1}".format(prefix, key))
 
-
     def _build_db_metrics(self, data, tags):
         for key, value in data['sizes'].items():
             self.gauge("couchdb.by_db.{0}_size".format(key), value, tags)
 
         for key in ['purge_seq', 'doc_del_count', 'doc_count']:
             self.gauge("couchdb.by_db.{0}".format(key), data[key], tags)
 
+    def _build_dd_metrics(self, info, tags):
+        data = info['view_index']
+        ddtags = list(tags)
+        ddtags.append("dd:{0}".format(info['name']))
+        ddtags.append("language:{0}".format(data['language']))
+
+        for key, value in data['sizes'].items():
+            self.gauge("couchdb.by_dd.{0}_size".format(key), value, ddtags)
```

Comment:
I think this metric prefix would be clearer if it were something along the lines of `couchdb.design_doc` / `couchdb.ddoc` - or even `couchdb.by_ddoc`